cmd/controller/pkg_repository_reconciler.go

```diff
@@ -66,6 +75,68 @@ func (r *PkgRepositoryReconciler) Reconcile(ctx context.Context, request reconci
 	return crdApp.Reconcile(force)
 }
 
+// ReconcileFetchPlaceholderSecrets helps determine if a placeholder secret
+// needs to be created for the PackageRepository. This placeholder secret is
+// populated by secretgen-controller so PackageRepositories can authenticate
+// to private registries without needing to explicitly declare a secretRef.
+// If no secretRef is specified for the PackageRepository, the placeholder
+// is created and used by the PackageRepository.
+func (r *PkgRepositoryReconciler) ReconcileFetchPlaceholderSecrets(pkgr v1alpha12.PackageRepository, app *v1alpha1.App) error {
+	for i, fetch := range app.Spec.Fetch {
+		if fetch.ImgpkgBundle != nil && fetch.ImgpkgBundle.SecretRef == nil {
+			secretName, err := r.createSecretForSecretgenController(pkgr, i)
+			if err != nil {
+				return err
+			}
+			app.Spec.Fetch[i].ImgpkgBundle.SecretRef = &v1alpha1.AppFetchLocalRef{secretName}
+		}
+
+		if fetch.Image != nil && fetch.Image.SecretRef == nil {
+			secretName, err := r.createSecretForSecretgenController(pkgr, i)
+			if err != nil {
+				return err
+			}
+			app.Spec.Fetch[i].Image.SecretRef = &v1alpha1.AppFetchLocalRef{secretName}
+		}
+	}
+	return nil
+}
+
+func (r *PkgRepositoryReconciler) createSecretForSecretgenController(pkgr v1alpha12.PackageRepository, i int) (string, error) {
```

Comment:
nit: what was the name we were using in the app package for this?

Reply:
Do you mean packageinstall package? It was also named `createSecretForSecretgenController`.

Reply:
oh ok. i thought it was renamed to Placeholder something nm.